src/test/test.spec.js

```diff
@@ -0,0 +1,235 @@
+import app from '../app';
+
+chai.use(chaiHttp);
+
+describe('SendIT API TESTS', () => {
+  // Test Invalid pathss
+  describe('/GET INVALID_PATH', () => {
+    it('should return not found', () => chai.request(app)
+      .get('/INVALID_PATH')
+      .then((res) => {
+        expect(res).to.have.status(404);
+      }));
+  });
+
+  // Test Parcels' requests
+  describe('Parcels', () => {
+    // Create parcel
+    describe('POST/ Parcel', () => {
+      // Parcel should be created
+      it('should create a parcel', () => {
+        const parcel = {
+          userId: 1,
+          parcel: {
+            parcelId: 1,
+            weigth: 40,
+            height: 80,
+            width: 40,
+            length: 90,
+            From: 'Kigali',
+            stateFrom: 'Kicukiro',
+            To: 'Rusizi',
+            stateTo: 'Muhondo',
+            services: 'Express',
+            status: 'delivered',
+          },
+        };
+        chai.request(app)
+          .post('/api/v1/parcels/')
+          .send(parcel)
+          .then((res) => {
+            expect(res).to.have.status(201);
+            expect(res.body).to.be.an('object');
+            expect(res.body).to.have.property('success').eql(true);
+            expect(res.body).to.have.property('parcelId');
+            expect(res.body).to.have.property('price');
+          });
+      });
+
+      // ASSIGN parcel to no exist User
+      it('should return user not exit', () => {
+        const parcel = {
+          userId: 'INVALID_USER_ID',
+          parcel: {
+            parcelId: 1,
+            weigth: 40,
+            height: 80,
+            width: 40,
+            length: 90,
+            From: 'Kigali',
+            stateFrom: 'Kicukiro',
+            To: 'Rusizi',
+            stateTo: 'Muhondo',
+            services: 'Express',
+            status: 'delivered',
+          },
+        };
+        chai.request(app)
+          .post('/api/v1/parcels/')
+          .send(parcel)
+          .then((res) => {
+            expect(res).to.have.status(404);
+            expect(res.body).to.be.an('object');
+            expect(res.body).to.have.property('success').eql(false);
+          });
+      });
+    });
+
+    // Get all parcels
+    describe('GET/ Parcels', () => {
+      it('should return all parcels', () => chai.request(app)
+        .get('/api/v1/parcels')
+        .then((res) => {
+          expect(res).to.have.status(200);
+          expect(res.body).to.be.an('object');
+          expect(res.body.parcels).to.be.an('array');
+        }));
+    });
+
+    // Get a Specific Parcel
+    describe('GET/ Specif Parcel', () => {
+      // Parcel Should be canceled
+      it('should return a parcel', () => chai.request(app)
+        .get('/api/v1/parcels/1')
+        .then((res) => {
+          expect(res).to.have.status(200);
+          expect(res.body).to.have.property('success').eql(true);
+          expect(res.body.parcel).to.be.an('object');
```

Comment:
On top of testing whether the body contains a parcel object, you should also test for the presence and correctness of the rest of the parcels properties.